thread_pool.h

```diff
@@ -55,7 +56,7 @@ class ThreadPool {
         }
 
         for(int i = 0 ; i < num_active_threads ; i++){
-            task_queue.emplace_back([]() {});
+            task_queue.emplace([]() {});
```

Comment:
Why is this changed?

Reply:
There is no queue.emplace_back method. Should I change task_queue to a vector ?

Reply:
Should you?

Reply:
No we do not need a vector.
This was changed for 2 reasons : 
First as i already said there is no queue.emplace_back() method
And second this method does the exact same thing as the emplace_back method for vectors